### PR TITLE
Clarified use with null or undefined strings

### PR DESCRIPTION
**Pull request by whorvath2**
### Description

Clarified that `test` works whether or not the supplied string is `undefined` or `null`.

### Motivation

Because I'm not a JS expert, I had to test this functionality manually. Having it documented here would have saved me time. Also, knowing that `test` works with uninitialized or `null` strings has value in helping users avoid both NPEs and throwing unnecessary exceptions. 

### Additional details

None.

### Related issues and pull requests

None.

### CODE REVIEW

**Review by Josh-Cena — COMMENTED** [review]
Hi, there's nothing special about `undefined` or `null`. They simply get coerced to strings.

```js
/undefined/.test(undefined); // true
/null/.test(null); // true
/1/.test(1); // true
```

What you could instead do is revert all changes and use a pattern similar to here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

> - `str`
>   - : The string against which to match the regular expression. All values are [coerced to strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#string_coercion), so omitting it or passing `undefined` causes `test()` to search for the string `"undefined"`, which is rarely what you want.

A simple sentence under the parameter description is enough. We really don't want to split hairs over what we already tell you is not a good idea as input (anything that's not a string).

**Comment by whorvath2** [comment]
@Josh-Cena I see what you're saying (and that's a reasonable solution), but I'll quibble a bit over your assertion: I don't see anything on this page that says using `test` against variables that might be `undefined`, `null`, or non-string variables is a bad idea, and that's kinda the point. I suggested these changes not only because it helps people like me (who understand the problem of NPEs but are less familiar with javascript) get the information we need to make informed decisions, but also to help less experienced programmers learn why this kind of thing matters, such as when you're not certain of the provenance of the values you're handling in your functions.

**Comment by Josh-Cena** [comment]
`null` is a separate data type in JavaScript that's unrelated to `string`; I don't think we have to teach this on an API reference page. So when we tell you that `str` is "a string", we naturally mean that anything other than string does not guarantee well-formed behavior. In Haskell terms (because in Java/C `null` is just a pointer):

```haskell
test :: String -> Bool
a = test "foo" -- works
b = test Nothing -- error
```

In my suggestion above, I did explicitly advise avoiding other inputs by saying "so omitting it or passing `undefined` causes `test()` to search for the string `"undefined"`, which is rarely what you want." The link to string coercion should also be sufficient to cover the behavior of non-string arguments.

File: files/en-us/web/javascript/reference/global_objects/regexp/test/index.md [hunk 1]
```diff
@@ -39,9 +40,11 @@ test(str)
...
 
 Use `test()` whenever you want to know whether a pattern is found in a
-string. `test()` returns a boolean, unlike the
-{{jsxref("String.prototype.search()")}} method (which returns the index of a match, or
-`-1` if not found).
+string. `test()` returns a boolean, unlike the {{jsxref("String.prototype.search()")}} 
+method (which returns the index of a match, or `-1` if a match is not found.) Using 
+`test()` is advantageous in cases when it's immaterial if the string to be 
+evaluated is `null` or `undefined`, because it will simply return `false` rather than
+throwing an error.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/regexp/test/index.md:47]: I don't know any string-handling method that throws an error when `null`/`undefined` is passed.

**Review by Josh-Cena — APPROVED** [review]
Instead of letting it sit around forever, I've applied my suggestions to make it align with other pages on MDN. Thanks for your contribution @whorvath2 and welcome to MDN :)

**Comment by whorvath2** [comment]
Thanks Josh! I went on vacation last week and wasn't able to handle this in time. I appreciate the help. :)
